### DISCUSSION TITLE
Using `next/dynamic` for JavaScript function, `document` is not defined

### DISCUSSION THREAD

**Discussion by trupakufi** [vercel/next.js#28422] [Help] [answered]
The document is not defined when i use aos.init that library aos, uses document, and i see that document is not defined, when a use aos.init() in project.

![Captura de ecrã de 2021-08-22 21-38-00](https://user-images.githubusercontent.com/72455038/130369824-2e302e05-20df-42e7-a2e5-853c00a46ab8.png)
![Captura de ecrã de 2021-08-22 21-40-50](https://user-images.githubusercontent.com/72455038/130369828-9d956281-d42b-445f-86bf-a88fc64610a1.png)

**Comment by G3root** [comment, marked as answer]
`next/dynamic` only works with React components, for JavaScript functions you have to use **native** [dynamic import](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import#dynamic_imports).

here is an example for above scenario: https://stackblitz.com/edit/nextjs-dtvwex?file=pages/index.js
